osu.Game.Tests/Visual/Online/TestSceneChatOverlay.cs

```diff
@@ -84,18 +74,71 @@ public void TestCloseChannelWhileSelectorClosed()
         {
             AddStep("Join channel 1", () => channelManager.JoinChannel(channel1));
             AddStep("Join channel 2", () => channelManager.JoinChannel(channel2));
-
             AddStep("Switch to channel 2", () => clickDrawable(chatOverlay.TabMap[channel2]));
             AddStep("Close channel 2", () => clickDrawable(((TestChannelTabItem)chatOverlay.TabMap[channel2]).CloseButton.Child));
-
             AddAssert("Selector remained closed", () => chatOverlay.SelectionOverlayState == Visibility.Hidden);
             AddAssert("Current channel is channel 1", () => channelManager.CurrentChannel.Value == channel1);
-
             AddStep("Close channel 1", () => clickDrawable(((TestChannelTabItem)chatOverlay.TabMap[channel1]).CloseButton.Child));
+            AddAssert("Selector is visible", () => chatOverlay.SelectionOverlayState == Visibility.Visible);
+        }
 
+        [Test]
+        public void TestShowWhileLoading()
+        {
+            setupUnloadedChannelsTest();
+            fakeInitializeChat();
             AddAssert("Selector is visible", () => chatOverlay.SelectionOverlayState == Visibility.Visible);
         }
 
+        [Test]
+        public void TestHideWhileLoadingThenShow()
+        {
+            setupUnloadedChannelsTest();
+            AddStep("Close chat overlay", () => chatOverlay.Hide());
+            fakeInitializeChat();
+            AddAssert("Selector is still closed", () => chatOverlay.SelectionOverlayState == Visibility.Hidden);
+        }
+
+        [Test]
+        public void TestShowWithChannelsJoined()
+        {
+            AddStep("Hide chat", () => chatOverlay.Hide());
+            AddStep("Join channel 1", () => channelManager.JoinChannel(channel1));
+            AddStep("Join channel 2", () => channelManager.JoinChannel(channel2));
+            AddStep("Show chat", () => chatOverlay.Show());
+            AddAssert("Selector is still closed", () => chatOverlay.SelectionOverlayState == Visibility.Hidden);
+        }
+
+        [Test]
+        public void TestShowWhileLoadingWithJoined()
+        {
+            AddStep("Join channel 2", () => channelManager.JoinChannel(channel2));
+            setupUnloadedChannelsTest();
+            fakeInitializeChat();
+            AddAssert("Selector is still closed", () => chatOverlay.SelectionOverlayState == Visibility.Hidden);
+        }
+
+        private void fakeInitializeChat()
```

Comment:
no need to call it "fake". name it to explain what it's doing
also share the code with the main SetUp method, which is basically doing the same thing anyway.